test/astarte_core/cql_utils_test.exs

```diff
@@ -148,4 +149,42 @@ defmodule CQLUtilsTest do
     assert Astarte.Core.CQLUtils.endpoint_id("com.foo", 2, "/a/%{something}/foo") !=
              Astarte.Core.CQLUtils.endpoint_id("com.foo", 2, "/a/%{something}/bar")
   end
+
+  test "realm name does not get encoded" do
+    assert CQLUtils.realm_name_to_keyspace_name("example", "atestinstance") ==
+             "atestinstanceexample"
+
+    assert Realm.valid_name?(CQLUtils.realm_name_to_keyspace_name("example", "atestinstance")) ==
+             true
+  end
+
+  test "realm name is > 47 characters long" do
+    assert CQLUtils.realm_name_to_keyspace_name(
+             "averyveryverylongrealmnamejustforthistest",
+             "atestinstance"
+           ) ==
+             "yxrlc3rpbnn0yw5jzwf2zxj5dmvyexzlcnlsb25ncmvhbg1u"
+
+    assert Realm.valid_name?(
+             CQLUtils.realm_name_to_keyspace_name(
+               "averyveryverylongrealmnamejustforthistest",
+               "atestinstance"
+             )
+           ) == true
+  end
+
+  test "realm name is empty, instance id exists" do
+    assert CQLUtils.realm_name_to_keyspace_name("", "atestinstance") == "atestinstance"
+
+    assert Realm.valid_name?(CQLUtils.realm_name_to_keyspace_name("", "atestinstance")) ==
+             true
+  end
```

Comment:
The typespec says that the realm name should not be empty, I'd say to remove this test